src/mgmt/workflows/ftns/create.py

```diff
@@ -5,12 +5,28 @@
 
 from common.workflow import *
 from operators.ftns_operator import *
+from operators.chains_operator import *
+from operators.dfts_operator import *
+from operators.droplets_operator import *
 
+dfts_opr = DftOperator()
+chains_opr = ChainOperator()
 ftns_opr = FtnOperator()
+droplets_opr = DropletOperator()
 
 
-def ftn_create(ftn, name, body, spec):
+def ftn_create(task, ftn, name, body, spec):
     logger.info("Creating Ftn {}!".format(name))
     if not ftn:
         ftn = ftns_opr.get_stored_obj(name, spec)
+    if not dfts_opr.store.contains_obj(ftn.dft, KIND.dft):
+        task.raise_temporary_error(
+            "Parent DFT {} not yet created!".format(ftn.dft))
+    if not chains_opr.store.contains_obj(ftn.parent_chain, KIND.chain):
+        task.raise_temporary_error(
+            "Parent Chain {} not yet created!".format(ftn.parent_chain))
+    if len(droplets_opr.get_unallocated_droplets()) < 1:
+        task.raise_temporary_error(
+            "No droplets available for FTN")
+    droplets_opr.assign_droplet(ftn)
     return ftn
```

Comment:
each network node will offer 2 droplets, one is physically connects to tenant network, so can only used as fwd; the other physically connects to ZGC internal network, which can only used for ftn. So droplets do have an attribute (role) already decided by how they physically connected.
For example, when there are 3 nodes allocated for ZGC, we will have eth2(for example) of each node on tenant network, eth3 on ZGC internal network. so total we got 6 droplets, but up to 3 fwds (eth2s) and up to 3 ftns (eth3s)

you can leave it here and we can make adjustment once external API is hooked up.

Reply:
I see, so there's a restriction on the **type** of droplets that can be used for a FWD or FTN respectively.
Gotcha, lets revisit this when the external API is ready.